### PR TITLE
Make StaticClusterManager handle Replica_Unavailable server error

### PR DESCRIPTION
**Pull request by jsjtzyy**
- support handling Replica_Unavailable and Replica_Available in
StaticClusterManager
- introduce StateResourcePolicy into Replica
- add tests in StaticClusterManagerTest for verification

### CODE REVIEW

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/PartitionLayout.java [hunk 1]
```diff
@@ -220,15 +219,16 @@ private long getNewPartitionId() {
   }
 
   // Creates a Partition and corresponding Replicas for each specified disk
-  public Partition addNewPartition(List<Disk> disks, long replicaCapacityInBytes, String partitionClass) {
+  public Partition addNewPartition(List<Disk> disks, long replicaCapacityInBytes, String partitionClass,
```
Comment (vgkholla) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/PartitionLayout.java]: the `ClusterMapConfig` need not be passed in. You can get it in this class using `hardwareLayout.getClusterMapConfig()`

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/Replica.java [hunk 2]
```diff
@@ -29,27 +31,48 @@
...
+    } catch (Exception e) {
+      logger.error("Error creating resource state policy when instantiating a replica " + e);
+      throw new IllegalStateException("Error creating resource state policy when instantiating a replica " + partition,
+          e);
+    }
     validate();
   }
 
   Replica(HardwareLayout hardwareLayout, Partition partition, JSONObject jsonObject) throws JSONException {
     this.partition = partition;
```
Comment (vgkholla) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/Replica.java]: if possible, call the other constructor in this one
`this(partition, hardwareLayout.findDisk(jsonObject.getString("hostname"), jsonObject.getInt("port"), jsonObject.getString("mountPath")),  hardwareLayout.getClusterMapConfig())`

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/StaticClusterManager.java [hunk 3]
```diff
@@ -238,8 +239,9 @@ Disk getDiskWithMostUnallocatedRawCapacity(DataNode node, long minCapacity) {
     return maxCapacityDisk;
   }
 
-  PartitionId addNewPartition(List<Disk> disks, long replicaCapacityInBytes, String partitionClass) {
```
Comment (vgkholla) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/StaticClusterManager.java]: same here - the config should already be available

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/PartitionLayout.java (hunk 1 above)
Comment (jsjtzyy) [reply ambry-clustermap/src/main/java/com.github.ambry.clustermap/PartitionLayout.java]: yeah, you are right. Let me check all the places and remove unnecessary `ClusterMapConfig`.

**Comment by jsjtzyy** [comment]
addressed comments from @vgkholla .
./gradlew build && ./gradlew test succeeded.

**Pull request merged**